core/src/services/hdfs_native/writer.rs

```diff
@@ -15,28 +15,41 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use bytes::Buf;
+use bytes::Bytes;
 use hdfs_native::file::FileWriter;
 
-use crate::raw::oio;
+use crate::raw::*;
+use crate::services::hdfs_native::error::parse_hdfs_error;
 use crate::*;
 
 pub struct HdfsNativeWriter {
-    _f: FileWriter,
+    f: FileWriter,
 }
 
 impl HdfsNativeWriter {
     pub fn new(f: FileWriter) -> Self {
-        HdfsNativeWriter { _f: f }
+        HdfsNativeWriter { f }
     }
 }
 
 impl oio::Write for HdfsNativeWriter {
-    async fn write(&mut self, _bs: Buffer) -> Result<()> {
-        todo!()
+    async fn write(&mut self, mut bs: Buffer) -> Result<()> {
+        while bs.has_remaining() {
```

Comment:
`Bytes::copy_from_slice` will copy the entire buffer while writing, it will add much cost here. How about this:

```
while let Some(bs) = bs.next() {
    self.f.write(bs).await
}
```